pkg/controller.go

```diff
@@ -259,6 +269,26 @@ func (c *controller) getPodMemoryUsageThreshold(pod *corev1.Pod) float64 {
 	return threshold
 }
 
+func appNameLabel(pod *corev1.Pod) string {
+	if pod == nil {
```

Comment:
I wouldn't guard again pod being nil. If pod is nil, the code would crash where this method is called already. Since this is only used for metrics, maybe we could both the two methods in metrics_exporter.go and just pass to pod to `RecordPodEviction` ?